src/app/handler/cart.js

```diff
@@ -0,0 +1,468 @@
+                        callback(null, 'Team Fortress 2\'s item server may be down or Steam may be experiencing temporary connectivity issues');
+                    } else {
+                        callback(null, 'An error occurred while sending the offer (' + TradeOfferManager.EResult[err.eresult] + ')');
+                    }
+                    return;
+                }
+            }
+
+            return callback(err);
+        });
+    });
+};
+
+function createAlteredMessage (steamID, alteredItems) {
+    const noneAvailable = { our: 'I don\'t have any', their: 'You don\'t have any' };
+    const someAvailable = { our: 'I only have', their: 'You only have' };
+
+    const none = { our: [], their: [] };
+    const some = { our: [], their: [] };
+
+    const cart = getCart(steamID);
+
+    ['our', 'their'].forEach((whose) => {
+        alteredItems[whose].forEach((sku) => {
+            const name = schemaManager.schema.getName(SKU.fromString(sku));
+            if (cart.amount(sku, whose)) {
+                some[whose].push({ name, amount: cart.amount(sku, whose) });
+            } else {
+                none[whose].push(name);
+            }
+        });
+    });
+
+    ['our', 'their'].forEach((whose) => {
+        none[whose].forEach((name, i) => {
+            const last = (i === none[whose].length-1 && i > 0);
+
+            if (last) {
+                noneAvailable[whose] = noneAvailable[whose].slice(0, -1);
+                noneAvailable[whose] += ' or';
+            }
+
+            noneAvailable[whose] += ' ' + pluralize(name, 0);
+
+            if (!last && i > 0) {
+                noneAvailable[whose] += ',';
+            }
+        });
+    });
+
+    ['our', 'their'].forEach((whose) => {
+        some[whose].forEach((obj, i) => {
+            const name = obj.name;
+            const amount = obj.amount;
+
+            const last = (i === some[whose].length-1 && i > 0);
+
+            if (last) {
+                someAvailable[whose] = someAvailable[whose].slice(0, -1);
+                someAvailable[whose] += ' and';
+            }
+
+            someAvailable[whose] += ' ' + pluralize(name, amount, true);
+
+            if (!last && i > 0) {
+                someAvailable[whose] += ',';
+            }
+        });
+    });
+
+    let message = '';
+
+    ['our', 'their'].forEach((whose) => {
+        ['none', 'some'].forEach((which, i) => {
+            if (i) {
+                if (some[whose].length) {
+                    message += someAvailable[whose] + '\n';
+                }
+            } else {
+                if (none[whose].length) {
+                    message += noneAvailable[whose] + '\n';
+                }
+            }
+        });
+    });
+
+    return message;
+}
+
+exports.stringify = function (steamid) {
+    const cart = getCart(steamid);
+
+    if (!cart) {
+        return 'Your cart is empty';
+    }
+
+    if (cart.isEmpty()) {
+        return 'Your cart is empty';
+    }
```

Comment:
Double?

Reply:
If there is no cart, or it is empty, then it is empty

Reply:
Why not combine them with ||

Reply:
Wait, nope. Cant do that if cart doesn't exist B)

Reply:
You can, I'll make a commit